components/camera/camera_test.go

```diff
@@ -223,6 +223,31 @@ func TestReconfigurableCamera(t *testing.T) {
 	test.That(t, stream.Close(context.Background()), test.ShouldBeNil)
 }
 
+func TestSourceFromCamera(t *testing.T) {
+	videoSrc := &simpleSource{"rimage/board1_small"}
+	actualCam, err := camera.NewFromReader(context.Background(), videoSrc, nil, camera.UnspecifiedStream)
+	test.That(t, err, test.ShouldBeNil)
+
+	reconfCam, err := camera.WrapWithReconfigurable(actualCam, resource.Name{})
+	test.That(t, reconfCam, test.ShouldNotBeNil)
+	test.That(t, err, test.ShouldBeNil)
+	source, err := camera.SourceFromCamera(reconfCam.(camera.Camera))
```

Comment:
You should probably test that `SourceFromCamera` returns `actualCam` here and below.